packages/zoe/tools/priceAuthorityRegistry.js

```diff
@@ -96,76 +105,118 @@ export const makePriceAuthorityRegistry = () => {
+          amountOut,
+        );
+      },
+      makeQuoteNotifier(amountIn, brandOut) {
+        return E(paFor(amountIn.brand, brandOut)).makeQuoteNotifier(
+          amountIn,
+          brandOut,
+        );
+      },
+      quoteAtTime(deadline, amountIn, brandOut) {
+        return E(paFor(amountIn.brand, brandOut)).quoteAtTime(
+          deadline,
+          amountIn,
+          brandOut,
+        );
+      },
+      quoteWhenLT: makeQuoteWhen('LT'),
+      quoteWhenLTE: makeQuoteWhen('LTE'),
+      quoteWhenGTE: makeQuoteWhen('GTE'),
+      quoteWhenGT: makeQuoteWhen('GT'),
+      mutableQuoteWhenLT: makeMutableQuoteWhen('LT'),
+      mutableQuoteWhenLTE: makeMutableQuoteWhen('LTE'),
+      mutableQuoteWhenGTE: makeMutableQuoteWhen('GTE'),
+      mutableQuoteWhenGT: makeMutableQuoteWhen('GT'),
     },
-    async quoteWanted(brandIn, amountOut) {
-      return E(paFor(brandIn, amountOut.brand)).quoteWanted(brandIn, amountOut);
-    },
-    async makeQuoteNotifier(amountIn, brandOut) {
-      return E(paFor(amountIn.brand, brandOut)).makeQuoteNotifier(
-        amountIn,
-        brandOut,
-      );
-    },
-    async quoteAtTime(deadline, amountIn, brandOut) {
-      return E(paFor(amountIn.brand, brandOut)).quoteAtTime(
-        deadline,
-        amountIn,
-        brandOut,
-      );
-    },
-    quoteWhenLT: makeQuoteWhen('LT'),
-    quoteWhenLTE: makeQuoteWhen('LTE'),
-    quoteWhenGTE: makeQuoteWhen('GTE'),
-    quoteWhenGT: makeQuoteWhen('GT'),
-    mutableQuoteWhenLT: makeMutableQuoteWhen('LT'),
-    mutableQuoteWhenLTE: makeMutableQuoteWhen('LTE'),
-    mutableQuoteWhenGTE: makeMutableQuoteWhen('GTE'),
-    mutableQuoteWhenGT: makeMutableQuoteWhen('GT'),
-  });
+  );
 
   /** @type {PriceAuthorityRegistryAdmin} */
-  const adminFacet = Far('price authority admin facet', {
-    registerPriceAuthority(pa, brandIn, brandOut, force = false) {
-      /** @type {MapStore<Brand, PriceAuthorityRecord>} */
-      let priceStore;
-      if (assetToPriceStore.has(brandIn)) {
-        priceStore = assetToPriceStore.get(brandIn);
-      } else {
-        priceStore = makeScalarMapStore('brandOut');
-        assetToPriceStore.init(brandIn, priceStore);
-      }
+  const adminFacet = prepareExo(
+    baggage,
+    'price authority admin facet',
+    M.interface('priceAuthorityRegistryAdmin', {
+      registerPriceAuthority: M.callWhen(
+        M.await(M.remotable('priceAuthority')),
+        BrandShape,
+        BrandShape,
+      )
+        .optional(M.boolean())
+        .returns(M.remotable('deleter')),
+    }),
+    {
+      registerPriceAuthority(pa, brandIn, brandOut, force = false) {
+        /** @type {MapStore<Brand, PriceAuthorityRecord>} */
+        let priceStore;
+        if (assetToPriceStore.has(brandIn)) {
+          priceStore = assetToPriceStore.get(brandIn);
+        } else {
+          priceStore = provideDurableMapStore(baggage, 'brandOut');
+          assetToPriceStore.init(brandIn, priceStore);
+        }
```

Comment:
this looks like the provide pattern. use `provide()`?

Reply:
indeed! done